File: Problem01/Problem01.py
input = open("input.txt", "r")
numsStr = input.readlines()
nums = [int(i) for i in numsStr]

def findGreaterSingle(nums):
    result = 0
    for i in range(1, len(nums)):
        if (nums[i] > nums[i-1]):
            result += 1
    return result

def findGreaterTriple(nums):
    result = 0
    sum1 = 0;
    sum2 = 0;
    for i in range(3, len(nums)):
        sum1 = nums[i-3] + nums[i-2] + nums[i-1]
        sum2 = nums[i-2] + nums[i-1] + nums[i]
        if (sum2 > sum1):
            result += 1;
    return result

print("Part 1: " + str(findGreaterSingle(nums)) )
print("Part 2: " + str(findGreaterTriple(nums)) )

File: Problem01/Problem01.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class Problem01 {
  //================================================================================
  // TESTER
  //================================================================================
  public static void main(String[] args) {
    ArrayList<Integer> nums = new ArrayList<Integer>();

    try {
      File file = new File(args[0]);
      Scanner input = new Scanner(file);
      while (input.hasNextInt()) {
        nums.add(input.nextInt());
      }
      input.close();
    } catch (FileNotFoundException ex) {
      System.out.println("File not found!");
    }

    try {
      String mode = args[1];
      if (mode.equals("1")) {
        System.out.println(findGreaterSingle(nums));
      } else if (mode.equals("2")) {
        System.out.println(findGreaterTriple(nums));
      } else {
        System.out.println(-1);
      }
    } catch (ArrayIndexOutOfBoundsException e) {
      System.out.println("Invalid syntax: use \"java Problem01 [filename] [1/2]\".");
    }
  }

  //================================================================================
  // PART 1
  //================================================================================
  private static int findGreaterSingle(ArrayList<Integer> nums) {
    int result = 0;
    for (int i = 1; i < nums.size(); i++) {
      if (nums.get(i) > nums.get(i-1)) result++;
    }
    return result;
  }

  //================================================================================
  // PART 2
  //================================================================================
  private static int findGreaterTriple(ArrayList<Integer> nums) {
    int result = 0;
    int sum1, sum2;
    for (int i = 3; i < nums.size(); i++) {
      sum1 = nums.get(i-3)+nums.get(i-2)+nums.get(i-1);
      sum2 = nums.get(i-2)+nums.get(i-1)+nums.get(i);
      if (sum2 > sum1) result++;
    }
    return result;
  }
}
